### PR TITLE
Create a FilterAdminCommandsFeature

### PR DESCRIPTION
**Pull request by magicus**
This will hide unusable admin commands such as `/wynnproxy` etc from tab expand.

### CODE REVIEW

File: common/src/main/java/com/wynntils/features/user/FilterAdminCommandsFeature.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+        RootCommandNode<SharedSuggestionProvider> newRoot = new RootCommandNode<>();
+        for (CommandNode<SharedSuggestionProvider> child : root.getChildren()) {
+            if (!FILTERED_COMMANDS.contains(child.getName())) {
+                newRoot.addChild(child);
+            }
+        }
+
+        // We also need to add the arguments for all commands
+        newRoot.addChild(root.getChild("args"));
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/features/user/FilterAdminCommandsFeature.java]: What does this do?

Comment (magicus) [reply common/src/main/java/com/wynntils/features/user/FilterAdminCommandsFeature.java]: All command literals have a hardcoded dependency on an "args" string argument node. I assume this is something setup by whatever mod they have to inject commands like this (bungee, perhaps). 

In theory, I should recurse over all child nodes and see if they have more argument types to be added. At least, that's what I assume I'm supposed to do. But I could not be bothered. Should Wynncraft drastically change how their admin commands are provided, then the user need to disable this feature until we can fix it.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/FilterAdminCommandsFeature.java]: (keep in mind this is only command suggestions, so even if we filter out something we should not, the user doesn't lose much.)

File: common/src/main/java/com/wynntils/features/user/FilterAdminCommandsFeature.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            "wcl",
+            "wynnproxy");
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    public void onCommandPacket(CommandsPacketEvent event) {
+        RootCommandNode<SharedSuggestionProvider> root = event.getRoot();
+
+        RootCommandNode<SharedSuggestionProvider> newRoot = new RootCommandNode<>();
+        for (CommandNode<SharedSuggestionProvider> child : root.getChildren()) {
+            if (!FILTERED_COMMANDS.contains(child.getName())) {
```
Comment (HighCrit) [inline common/src/main/java/com/wynntils/features/user/FilterAdminCommandsFeature.java:42]: Wouldn't a set be more performant?

**Review by HighCrit — CHANGES_REQUESTED** [review]

File: common/src/main/java/com/wynntils/features/user/FilterAdminCommandsFeature.java (hunk 2 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/features/user/FilterAdminCommandsFeature.java:42]: Maybe. I doubt you can even measure it.

The Rules of Optimizations:
1) Don't Optimize
2) (Only for experts) Don't Optimize Yet.

:-)

But I can agree that it makes things a bit clearer (and that's worth more than premature optimization) if you call things a Set when you just need Set-like behavior.

Comment (magicus) [reply common/src/main/java/com/wynntils/features/user/FilterAdminCommandsFeature.java:42]: ... only there is no `asSet`. :( At least not that I can find.

File: common/src/main/java/com/wynntils/features/user/FilterAdminCommandsFeature.java (hunk 1 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/features/user/FilterAdminCommandsFeature.java]: It seemed like I could remove it without any bad effects. I don't fully understand the Brigadier structure. But if it works without it, I'll skip it.

File: common/src/main/java/com/wynntils/features/user/FilterAdminCommandsFeature.java (hunk 2 above)
Comment (HighCrit) [reply common/src/main/java/com/wynntils/features/user/FilterAdminCommandsFeature.java:42]: > ... only there is no `asSet`. :( At least not that I can find.

`new HashSet<String>(Arrays.asList(...))` is the best I could find 👀

**Review by HighCrit — APPROVED** [review]

**Pull request merged**